x-pack/plugins/file_upload/public/components/geo_upload_form/geo_upload_form.tsx

```diff
@@ -119,13 +141,39 @@ export class GeoUploadForm extends Component<Props, State> {
         <GeoFilePicker onSelect={this._onFileSelect} onClear={this._onFileClear} />
         {this._renderGeoFieldTypeSelect()}
         {this.state.hasFile ? (
-          <IndexNameForm
-            indexName={this.props.indexName}
-            indexNameError={this.props.indexNameError}
-            onIndexNameChange={this.props.onIndexNameChange}
-            onIndexNameValidationStart={this.props.onIndexNameValidationStart}
-            onIndexNameValidationEnd={this.props.onIndexNameValidationEnd}
-          />
+          <>
+            <IndexNameForm
+              indexName={this.props.indexName}
+              indexNameError={this.props.indexNameError}
+              onIndexNameChange={this.props.onIndexNameChange}
+              onIndexNameValidationStart={this.props.onIndexNameValidationStart}
+              onIndexNameValidationEnd={this.props.onIndexNameValidationEnd}
+            />
+            <EuiSpacer size="m" />
+            <EuiFormRow
+              label={
+                <EuiToolTip
+                  anchorClassName="eui-alignMiddle"
+                  content={i18n.translate('xpack.fileUpload.uploadSize.tooltip', {
+                    defaultMessage: 'Use small upload size to alleviate request timeout failures.',
+                  })}
+                >
+                  <span>
+                    {i18n.translate('xpack.fileUpload.uploadSize.label', {
+                      defaultMessage: 'Upload size',
+                    })}{' '}
+                    <EuiIcon type="questionInCircle" color="subdued" />
+                  </span>
+                </EuiToolTip>
+              }
+            >
+              <EuiSelect
```

Comment:
nit: IMO, the toggle switch is a better UI than the select box. How about using a switch with a label like "Upload file in smaller chunks" and modify the tooltip text accordingly?

Reply:
Nice suggestion. I have changed the UI back to a switch.